### ISSUE TITLE
docker emqx (etcd) cannot join cluster

### ISSUE THREAD

**Issue by seqwait** [emqx/emqx#9159]
### What happened?

after someone's node oom killed by the system, it can not join the cluster error log
```
2022-10-15 23:32:47.236 [info] [Plugins] Load plugin emqx_recon successfully
['2022-10-15T23:32:49Z']:waiting emqx
['2022-10-15T23:32:50Z']:waiting emqx
['2022-10-15T23:32:52Z']:waiting emqx
['2022-10-15T23:32:53Z']:waiting emqx
['2022-10-15T23:32:55Z']:waiting emqx
['2022-10-15T23:32:55Z']:timeout error
```

### What did you expect to happen?

join the cluster success

### Anything else we need to know?

docker version: Docker version 19.03.8, build afacb8b
docker network : host
autocluster based on etcd

### EMQX version

<details>

```console
sysdescr  : EMQ X Broker
version   : 4.0.0
uptime    : 472 days, 16 hours, 25 minutes, 27 seconds
datetime  : 2022-10-15 23:44:38
```

</details>

### OS version

<details>

```console
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
</details>

### Log files

<details>
['2022-10-15T23:32:49Z']:waiting emqx
['2022-10-15T23:32:50Z']:waiting emqx
['2022-10-15T23:32:52Z']:waiting emqx
['2022-10-15T23:32:53Z']:waiting emqx
['2022-10-15T23:32:55Z']:waiting emqx
['2022-10-15T23:32:55Z']:timeout error
</details>

**Comment by seqwait** [comment]
> @seqwait Are you really using 4.0.0 ?

yes, it same like the 4.0.0 restart order bug

**Comment by ieQu1** [comment]
The issue with node startup running into a deadlock after node restart has been fixed here: https://github.com/emqx/mria/blob/21c5954ff4db050be651d37e753f882d3b707813/src/mria_mnesia.erl#L262 (note that `wait_for_tables` does in a loop with timeout to prevent deadlocks). I believe this patch was merged to EMQX 4.3.
